pkg/runtime/runtime.go

```diff
@@ -1184,17 +1185,18 @@ func (a *DaprRuntime) initState(s components_v1alpha1.Component) error {
 		// set specified actor store if "actorStateStore" is true in the spec.
 		actorStoreSpecified := props[actorStateStore]
 		if actorStoreSpecified == "true" {
-			if a.actorStateStoreCount++; a.actorStateStoreCount == 1 {
+			a.actorStateStoreLock.Lock()
+			if a.actorStateStoreName == "" {
+				log.Infof("detected actor state store: %s", s.ObjectMeta.Name)
 				a.actorStateStoreName = s.ObjectMeta.Name
+			} else {
+				log.Warnf("ignoring duplicate actor state store: %s", s.ObjectMeta.Name)
 			}
+			a.actorStateStoreLock.Unlock()
 		}
 		diag.DefaultMonitoring.ComponentInitialized(s.Spec.Type)
 	}
 
-	if a.hostingActors() && (a.actorStateStoreName == "" || a.actorStateStoreCount != 1) {
```

Comment:
This code is executed before sidecar invokes /dapr/config, which means that a.hostingActors() would return false. This was not always like that but we had many changes in the order of loading things in Dapr sidecar and this was a side effect of that.